File: Java/RecyclerView/RecyclerView-mvvm/app/src/main/java/com/ls/project/recyclerview/viewmodel/MainViewModel.java
package com.ls.project.recyclerview.viewmodel;

import android.support.v7.widget.RecyclerView;

import com.ls.project.recyclerview.adapter.ListRecyclerViewAdapter;
import com.ls.project.recyclerview.model.ListItem;

import java.util.ArrayList;

public class MainViewModel implements ViewModel {
    public ListRecyclerViewAdapter mAdapter;
    private ArrayList<ListItem> items;

    @Override
    public void onCreate() {

        items = new ArrayList<>();
        for(int i =1;i<10;i++ ) {
            items.add(new ListItem("구역 "+String.valueOf(i),String.valueOf(i)+" 층",String.valueOf(i*10)));
        }
        mAdapter = new ListRecyclerViewAdapter(items);

    }

    @Override
    public void onDestroy() {

    }

    @Override
    public void onPause() {

    }

    @Override
    public void onResume() {

    }
}

File: RecyclerView/RecycelerView-mvvm/app/src/main/java/com/ls/project/areacard/viewmodel/MainViewModel.kt
package com.ls.project.areacard.viewmodel

import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import com.ls.project.areacard.adapter.ListRecyclerViewAdapter
import com.ls.project.areacard.model.ListItem

class MainViewModel : ViewModel {
    lateinit var adapter: ListRecyclerViewAdapter
    private lateinit var items: ArrayList<ListItem>
    override fun onCreate() {
        items = arrayListOf<ListItem>()
        for (i in 0..10) {
            items.add(ListItem("구역 $i", "${i * 10}"))
        }
        adapter = ListRecyclerViewAdapter(items)
        /** 2 */
//        adapter.setOnItemClickListener(object : ListRecyclerViewAdapter.ClickListener{
//            override fun onItemClick(v: View, pos: Int) {
//                var linear = (v as LinearLayout)
//                var text = "${getChildText(linear,0)} ${getChildText(linear,1)} ${getChildText(linear,2)}"
//                Toast.makeText(linear.context,text,Toast.LENGTH_SHORT).show()
//            }
//
//            override fun onItemLongClick(v: View, pos: Int) {
//                var linear = (v as LinearLayout)
//                var text = "${getChildText(linear,0)} ${getChildText(linear,1)} ${getChildText(linear,2)}"
//                Toast.makeText(linear.context,"$text Long",Toast.LENGTH_SHORT).show()
//            }
//        })

        /** 1 */
//        adapter.onClick = {
//            var text = "${getChildText(it, 0)} ${getChildText(it, 1)} ${getChildText(it, 2)}"
//            Toast.makeText(it.context, text, Toast.LENGTH_SHORT).show()
//        }

    }

    /** 1 */
//    private fun getChildText(view: View, pos: Int): String {
//        return ((view as LinearLayout).getChildAt(pos) as TextView).text.toString()
//    }

    override fun onDestroy() {
    }

    override fun onPause() {
    }

    override fun onResume() {
    }
}